src/serializer/ResidualFunctions.js

```diff
@@ -225,7 +225,12 @@ export class ResidualFunctions {
     for (let [funcBody, instances] of functionEntries) {
       let functionInfo = this.residualFunctionInfos.get(funcBody);
       invariant(functionInfo);
-      this._referentialize(functionInfo.unbound, instances);
+      let additionalFunctionInstances = instances.filter((instance) => rewrittenAdditionalFunctions.has(instance.functionValue))
+      if (additionalFunctionInstances && additionalFunctionInstances.length) {
+        invariant(additionalFunctionInstances.length === instances.length, additionalFunctionInstances.length);
```

Comment:
I'll make it into a `TODO:` error. I can probably support it with a bit of refactoring.